renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -37,13 +38,418 @@ constexpr VkDeviceSize handleCountSize = 8;
+        VkAccelerationStructureBuildRangeInfoKHR &buildData = geoData.buildRangeInfo;
+        buildData.primitiveCount = rangeInfo.primitiveCount;
+        buildData.primitiveOffset = 0;
+        buildData.firstVertex = 0;
+        buildData.transformOffset = 0;
+
+        if(indexData)
+        {
+          buildData.primitiveOffset = (uint32_t)AlignUp(vertexData.size, vertexData.alignment);
+          buildData.firstVertex = rangeInfo.firstVertex;
+        }
+        if(transformData)
+          buildData.transformOffset = (uint32_t)(AlignUp(vertexData.size, vertexData.alignment) +
+                                                 AlignUp(indexData.size, indexData.alignment));
+
+        metadata->geometryData.push_back(geoData);
+
+        break;
+      }
+      case VK_GEOMETRY_TYPE_AABBS_KHR:
+      {
+        const VkAccelerationStructureGeometryAabbsDataKHR &aabbInfo = geometry.geometry.aabbs;
+
+        // Find the associated VkBuffer
+        BufferData data = GetDeviceAddressData(aabbInfo.data.deviceAddress);
+        if(!data)
+        {
+          RDCERR("Unable to find VkBuffer for AABB data at 0x%llx", aabbInfo.data.deviceAddress);
+          continue;
+        }
+
+        data.size = rangeInfo.primitiveCount * sizeof(VkAabbPositionsKHR);
+        data.SetReadPosition(rangeInfo.primitiveOffset);
+
+        // Get the alignment
+        VkMemoryRequirements mrq = {};
+        ObjDisp(device)->GetBufferMemoryRequirements(Unwrap(device), data.buf, &mrq);
+
+        // Allocate copy buffer
+        readbackmem = CreateReadBackMemory(device, data.size, mrq.alignment);
+        if(readbackmem.mem == VK_NULL_HANDLE)
+        {
+          RDCERR("Unable to allocate AS AABB input buffer readback memory (size: %u bytes)",
+                 mrq.size);
+          continue;
+        }
+
+        // Insert copy commands
+        VkBufferCopy region = {
+            data.GetReadPosition(),
+            0,
+            data.size,
+        };
+        ObjDisp(device)->CmdCopyBuffer(Unwrap(commandBuffer), data.buf, readbackmem.buf, 1, &region);
+
+        // Store the metadata
+        VkAccelerationStructureInfo::GeometryData geoData;
+        geoData.geometryType = geometry.geometryType;
+        geoData.flags = geometry.flags;
+        geoData.readbackMem = readbackmem.mem;
+        geoData.memSize = readbackmem.size;
+
+        geoData.aabbs.stride = aabbInfo.stride;
+
+        geoData.buildRangeInfo = rangeInfo;
+        geoData.buildRangeInfo.primitiveOffset = 0;
+
+        metadata->geometryData.push_back(geoData);
+
+        break;
+      }
+      case VK_GEOMETRY_TYPE_INSTANCES_KHR:
+      {
+        const VkAccelerationStructureGeometryInstancesDataKHR &instanceInfo =
+            geometry.geometry.instances;
+
+        if(instanceInfo.arrayOfPointers)
+          return RDResult(ResultCode::InternalError,
+                          "AS instance build arrayOfPointers unsupported");
```

Comment:
FYI there is a helper `RETURN_ERROR_RESULT` which can do this and also print the `RDCERR` for you, without the calling code needing to do that.